apps/lib/apps.c

```diff
@@ -3295,3 +3301,29 @@ EVP_PKEY *app_paramgen(EVP_PKEY_CTX *ctx, const char *alg)
                      opt_getprog(), alg != NULL ? alg : "asymmetric");
     return res;
 }
+
+/*
+ * Return non-zero if the legacy path is still an option.
+ * This decision is based on the global command line operations and the
+ * behaviour thus far.
+ */
+int opt_legacy_okay(void)
+{
+    int provider_options = opt_provider_option_given();
+    int libctx = app_get0_libctx() != NULL || app_get0_propq() != NULL;
+#ifndef OPENSSL_NO_ENGINE
+    ENGINE *e = ENGINE_get_first();
+
+    if (e != NULL) {
+        ENGINE_free(e);
+        return 1;
+    }
+#endif
+    /*
```

Comment:
I've a suspicion that from here down could just be replaced by `return 0;`.

I.e. if an engine isn't loaded, legacy paths are never required.

The conservative approach is to not do this just in case.

Reply:
Yes, this should not be needed if it is ok that engines added by command line -engine option will affect the  opt_legacy_okay() output only if the -engine is parsed before the opt_legacy_okay() is called.

Reply:
It would come down to when the configuration loads versus when this is called.  I think it best to leave this in for the moment.

Reply:
It would come down to when the configuration loads versus when this is called.  I think it best to leave this in for the moment.

Fixes pushed.